plugins/content/caption/base/lib/kCaptionsContentManager.php

```diff
@@ -89,4 +155,110 @@ public static function addParseMultiLanguageCaptionAssetJob($captionAsset, $file
 
 		return kJobsManager::addJob($batchJob, $jobData, $jobType);
 	}
+
+
+	public static function onTimeRange($captionStartTime, $captionEndTime, $clipStartTime, $clipEndTime)
+	{
+		//caption asset items which started before clip start time but ended after the clip started
+		if(($captionEndTime >= $clipStartTime) && ($captionStartTime <= $clipStartTime))
+			return true;
+
+		//caption asset items which started during clip time range
+		if (($captionStartTime >= $clipStartTime) && ($captionStartTime <= $clipEndTime))
+			return true;
+
+		return false;
+	}
+
+
+	public static function getAdjustedStartTime($captionStartTime, $clipStartTime)
+	{
+		$adjustedStartTime = $captionStartTime - $clipStartTime;
+		if ($adjustedStartTime < 0)
+			$adjustedStartTime = 0;
+		return $adjustedStartTime;
+	}
+
+
+	public static function getAdjustedEndTime($clipStartTime, $clipEndTime, $captionEndTime)
+	{
+		$adjustedEndTime = $captionEndTime - $clipStartTime;
+		if ($adjustedEndTime > $clipEndTime)
+			$adjustedEndTime = $clipEndTime;
+		return $adjustedEndTime;
+	}
+
+	/**
+	 * @param $timeStr
+	 * @return string
+	 */
+	public static function parseStrTTTime($timeStr)
+	{
+		$error = null;
+		$tabs = explode(':', $timeStr);
+		$result = count($tabs);
+		$timeInMilliseconds = null;
+		if ($result == 2)
+			$timeInMilliseconds = self::shortTimeFormatToInteger($timeStr) ;
+		elseif  ($result == 3)
+			$timeInMilliseconds = kXml::timeToInteger($timeStr);
+		else
+			$error = 'Error parsing time to milliseconds. invalid format for '.$timeStr;
+
+		return array ($timeInMilliseconds, $error);
+	}
+
+	private static function shortTimeFormatToInteger($time)
+	{
+		$parts = explode(':', $time);
+		if(!isset($parts[0]) || !is_numeric($parts[0]))
+			return null;
+
+		$ret = intval($parts[0]) * (60 * 1000);  // hours im milliseconds
+
+		if(!isset($parts[1]))
+			return $ret;
+		if(!is_numeric($parts[1]))
+			return null;
+
+		if(!isset($parts[1]))
+			return $ret;
+		if(!is_numeric($parts[1]))
+			return null;
+
+		$ret += floatval($parts[1]) * 1000;  // seconds im milliseconds
+		return round($ret);
+	}
+
+
+	public function createCaptionsFile($content, $clipStartTime, $clipEndTime, $timeCode)
```

Comment:
This function is being called as if it is static , since it access the instance, change all the callers to use it as non-static